vpr/src/draw/search_bar.cpp

```diff
@@ -254,45 +287,82 @@ void highlight_blocks(ClusterBlockId clb_index) {
     application.refresh_drawing();
 }
 
-void highlight_nets(ClusterNetId net_id) {
-    t_trace* tptr;
-    auto& route_ctx = g_vpr_ctx.routing();
+/**
+ * @brief Finds and highlights the atom block. Returns true if block shows, false if not
+ * 
+ * @param atom_blk AtomBlockId being searched for
+ * @param cl_blk ClusterBlock containing atom_blk
+ * @param app ezgl:: application used
+ * @return true | If sub-block can be highlighted
+ * @return false | If sub-block not found (impossible in search case) or not shown at current zoom lvl
+ */
+bool highlight_atom_block(AtomBlockId atom_blk, ClusterBlockId cl_blk, ezgl::application* app) {
+    auto& atom_ctx = g_vpr_ctx.atom();
+    auto& cl_ctx = g_vpr_ctx.clustering();
+    t_pb* pb = cl_ctx.clb_nlist.block_pb(cl_blk);
 
-    t_draw_state* draw_state = get_draw_state_vars();
+    //Getting the pb* for the atom block
+    auto atom_block_pb = find_atom_block_in_pb(atom_ctx.nlist.block_name(atom_blk), pb);
+    if (!atom_block_pb) return false; //If no block found, returning false
 
-    if (int(route_ctx.trace.size()) == 0) return;
+    //Ensuring that block is drawn at current zoom lvl, returning false if not
+    auto atom_block_depth = atom_block_pb->pb_graph_node->pb_type->depth;
+    t_draw_state* draw_state = get_draw_state_vars();
+    int max_depth = draw_state->show_blk_internal;
+    if (atom_block_depth > max_depth) return false;
 
-    for (tptr = route_ctx.trace[net_id].head; tptr != nullptr; tptr = tptr->next) {
-        draw_state->net_color[net_id] = ezgl::MAGENTA;
-    }
+    //Highlighting block
+    get_selected_sub_block_info().set(atom_block_pb, cl_blk);
+    app->refresh_drawing();
+    return true;
 }
 
-void highlight_nets(std::string net_name) {
-    auto& cluster_ctx = g_vpr_ctx.clustering();
-
-    ClusterNetId net_id = ClusterNetId::INVALID();
-    net_id = cluster_ctx.clb_nlist.find_net(net_name);
-
-    if (net_id == ClusterNetId::INVALID()) {
-        warning_dialog_box("Invalid Net Name");
-        return; //name not exist
+/**
+ * @brief Recursively looks through pb graph to find block w. given name
+ * 
+ * @param name name of block being searched for
+ * @param pb current node to be examined
+ * @return t_pb* t_pb ptr of block w. name "name"
+ */
+t_pb* find_atom_block_in_pb(std::string name, t_pb* pb) {
+    //Checking if block is one being searched for
+    std::string pbName(pb->name);
+    if (pbName == name) return pb;
```

Comment:
Prefer to have simple if statements like this broken over two lines.  Not sure if make format will do that.